test/node/test-buffer-badhex.js

```diff
@@ -14,6 +14,12 @@ const assert = require('assert');
   assert.strictEqual(buf.write('abcdef01', 0, 'hex'), 4);
   assert.deepStrictEqual(buf, new Buffer([0xab, 0xcd, 0xef, 0x01]));
   assert.strictEqual(buf.toString('hex'), 'abcdef01');
+  // Node Buffer behavior check
+  // > Buffer.from('abc def01','hex')
+  // <Buffer ab>
+  assert.strictEqual(buf.write('abc def01', 0, 'hex'), 1);
+  assert.deepStrictEqual(buf, new Buffer([0xab]));
+  assert.strictEqual(buf.toString('hex'), 'ab');
```

Comment:
`buf` here is a 4 bytes long buffer, so these asserts can never succeed? Did you try this locally? 🤔

Reply:
buf should NOT be a 4 byte buffer. That's the point.

If it is a 4 byte buffer, than the code is broken.

a non-hex character should end the hex parsing and only the 2-hex pairs up until the first non-hex is output as a Buffer.

in this case, `[0xab]`

Reply:
But you are doing `.write` on an already existing `Buffer` that is 4 bytes long. `write` cannot truncate the length of the buffer.

With my updated tests it checks that `write` only wrote 1 byte, and that it left the three bytes after as `00 00 00`.